### PR TITLE
Update the readme path to docs repo

### PR DESCRIPTION
**Pull request by sima-zhu**
Testing pipeline on daily: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1897102&view=logs&j=f8e040b3-c0ff-5789-0eda-99daaaa3fc1b&t=4ddc76a6-85f2-526e-fa3e-b1020f31ed12
Testing pipeline on release: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1895077&view=results
Commit: 
https://github.com/Azure/azure-docs-sdk-dotnet/commit/711e855aea1099ce15cda4d507772ef748da95ba
Update docs metadata · Azure/azure-docs-sdk-dotnet@711e855
Repository for documentation on the Azure .NET SDK. - Update docs metadata · Azure/azure-docs-sdk-dotnet@711e855

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -284,8 +284,8 @@ function Get-dotnet-DocsMsMetadataForPackage($PackageInfo) {
 
   New-Object PSObject -Property @{
     DocsMsReadMeName = $readmeName
-    LatestReadMeLocation = 'api/overview/azure'
-    PreviewReadMeLocation = 'api/overview/azure'
+    LatestReadMeLocation = 'api/overview/azure/latest'
+    PreviewReadMeLocation = 'api/overview/azure/preview'
     Suffix = $suffix
```
Comment (danieljurek) [inline eng/scripts/Language-Settings.ps1:289]: `Suffix` should be an empty string here. If not, releasing a preview library would result in package-level overview pages being named with `-pre.md`.

Comment (danieljurek) [reply eng/scripts/Language-Settings.ps1:289]: I think this can be directly tested by running a publish of the `template` pipeline.

File: eng/pipelines/templates/jobs/ci.yml [hunk 2]
```diff
@@ -214,7 +214,7 @@ jobs:
           /p:BuildSnippets=true
         displayName: Build snippets
         condition: and(succeeded(), eq(${{ parameters.BuildSnippets }}, true))
-      - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
+      # - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.yml:217]: Let's not forget to undo this change.

**Review by danieljurek — APPROVED** [review]
Changes look good. I'm seeing the template readme publishing in the `preview` folder and having the correct name (no `-pre.md` at the end of the filename) 

```
Already on 'daily/2022-10-05'
M	api/overview/azure/preview/template-readme.md
M	metadata/preview/Azure.Template.json
Your branch is up to date with 'origin/daily/2022-10-05'.
```

By contrast, running against the `main` branch gives the following results today: 

```
Already on 'daily/2022-10-05'
M	api/overview/azure/template-readme-pre.md
M	metadata/preview/Azure.Template.json
Your branch is up to date with 'origin/daily/2022-10-05'.
```

Publishing also looks good: 

```
Already on 'main'
A	api/overview/azure/preview/template-readme.md
M	metadata/preview/Azure.Template.json
Your branch is up to date with 'origin/main'.
```

An old release of template looks like: 

```
Already on 'main'
M	api/overview/azure/template-readme-pre.md
M	metadata/preview/Azure.Template.json
Your branch is up to date with 'origin/main'.
```


This can merge when it's appropriate given the other changes we have going into the .NET docs repo!

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 3]
```diff
@@ -178,8 +178,8 @@ stages:
                             TargetDocRepoName: ${{parameters.TargetDocRepoName}}
                             Language: 'dotnet'
                             SparseCheckoutPaths:
-                              - api/overview/azure/
-                              - metadata/
+                              - /api/overview/azure/
```
Comment (sima-zhu) [inline eng/pipelines/templates/stages/archetype-net-release.yml:181]: The change right here is to checkout path start from the prefix.
